packages/azure-kusto-ingest/src/ingestClient.browser.ts

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { KustoConnectionStringBuilder } from "azure-kusto-data";
+
+import { BlobDescriptor, StreamDescriptor } from "./descriptors";
+import { FileDescriptor } from "./fileDescriptor";
+import { QueueSendMessageResponse } from "@azure/storage-queue";
+
+import { IngestionPropertiesInput } from "./ingestionProperties";
+import { KustoIngestClientBase } from "./ingestClientBase";
+import { BlockBlobUploadOptions } from "@azure/storage-blob";
+
+export class KustoIngestClient extends KustoIngestClientBase {
+    constructor(kcsb: string | KustoConnectionStringBuilder, defaultProps?: IngestionPropertiesInput) {
+        super(kcsb, defaultProps, true);
+    }
+
+    /**
+     * Use string for Node.js and Blob for browser
+     */
+    async ingestFromFile(file: FileDescriptor | Blob | string, ingestionProperties?: IngestionPropertiesInput): Promise<QueueSendMessageResponse> {
+        this.ensureOpen();
+        if (!(file instanceof Blob) && !((file as FileDescriptor).file instanceof Blob)) {
+            throw new Error("Expected object of type Blob");
+        }
+        const descriptor = file instanceof FileDescriptor ? file : new FileDescriptor(file);
+
+        const extension = descriptor.extension || ingestionProperties?.format || "csv";
+        const blob = descriptor.file as Blob;
+        const props = this._getMergedProps(ingestionProperties);
+        const name = descriptor.name ? `__${descriptor.name}` : "";
+        const blobName = `${props.database}__${props.table}__${descriptor.sourceId}${name}.${extension}`;
```

Comment:
Can we remove some of the duplication here?

Reply:
only duplicate is 
 this.ensureOpen();
        const props = this._getMergedProps(ingestionProperties);
            const blockBlobClient = await this._getBlockBlobClient(blobName);
It doesnt seem to clearer if we extract it?

Reply:
ctrl +f ${props.database}__